### PR TITLE
BeamSpot + Rho variables in nanoAODs

### PR DESCRIPTION
**Pull request by maxgalli**
#### PR description:

This PR introduces the photon variables required to allow H->gamma gamma to move to nanoAODs.
The changes were discussed in [this](https://indico.cern.ch/event/1085967/#4-hgg-photon-updates-for-nano) XPOG meeting.

### EDIT

The photons related variables were moved to https://github.com/cms-sw/cmssw/pull/36526

### CODE REVIEW

**Comment by mariadalfonso** [comment]
please test

**Comment by gouskos** [comment]
please test

**Comment by mariadalfonso** [comment]
please test

**Comment by maxgalli** [comment]
> @maxgalli we need to add the extra variables in the DQM PhysicsTools/NanoAOD/python/nanoDQM_cfi.py

ok, I will take care soon

**Comment by mariadalfonso** [comment]
> * Reco comparison results: 138 differences found in the comparisons

There are differences in the photon and rho tables, touches by this PR.
but also in muon, electron, tau , isoTrack, rivetLepton to be checked if are related to the addition of int in the DataFormat

**Comment by mariadalfonso** [comment]
can you add also the rho and beamSpot ? Thanks

**Comment by gouskos** [comment]
please test

**Comment by maxgalli** [comment]
> /cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36037/20585/install.sh

most likely due to the way I added variables to the DQM for rho and beamspot, trying to fix it now

**Comment by maxgalli** [comment]
> > /cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36037/20585/install.sh
> 
> most likely due to the way I added variables to the DQM for rho and beamspot, trying to fix it now

I actually forgot the title key in two of the histograms, fixed now

**Comment by mariadalfonso** [comment]
please test

**Comment by mariadalfonso** [comment]
The range of the beam spot DQM is not well defined
Screen Shot 2021-11-26 at 14 56 02

**Comment by mariadalfonso** [comment]
Rho Changed from 5 separate float to a block with 6 floats

Screen Shot 2021-11-26 at 15 30 57

image

**Comment by mariadalfonso** [comment]
There are numerous changes for the non photons/Rho/BeamSpot listed here 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_2_X_2021-11-17-1100+7fdc3b/46913/validateJR/all_OldVSNew_RunJetHT2018CnanoEDM106Xv2wf136p8523/

Among them: (goes in the right direction to remove these  UChar_t )
NEW Electron_cutBased       Int_t   0.4     1.5             1.0
WAS Electron_lostHits      UChar_t 0.4     1.5             1.0%

once can see the dump here:

this PR
https://cms-nanoaod-integration.web.cern.ch/integration/36037/mc106Xul17v2_size_report.html
vs the reference 
https://cms-nanoaod-integration.web.cern.ch/integration/cms-swmaster/mc106Xul17v2_size_report.html

**Comment by mariadalfonso** [comment]
@maxgalli

This PR has conflict now.

To proceed with this PR we can do the following: move in a separate PR the photon changes (this add 8% of size so we should optimise separately) and keep this PR for rho and beamspot and related changes in Output class ( that is used for the beamspot)

in parallel: resolve conflict, fix ranges for DQM and of course understand why we have these unwanted changes https://github.com/cms-sw/cmssw/pull/36037#issuecomment-980024682

**Comment by mariadalfonso** [comment]
please test

**Comment by mariadalfonso** [comment]
> * Reco comparison results: 5706 differences found in the comparisons

we have now unrelated changes also in MINI-AOD
will retest later with an extra IB

**Comment by mariadalfonso** [comment]
please test

**Comment by mariadalfonso** [comment]
please test

**Comment by gouskos** [comment]
please test

File: DataFormats/NanoAOD/interface/FlatTable.h [hunk 1]
```diff
@@ -41,6 +41,7 @@ namespace nanoaod {
       Float,
       Int,
       UInt8,
+      Int8,
```
Comment (mariadalfonso) [inline DataFormats/NanoAOD/interface/FlatTable.h:44]: can you have this as last of the enum ?
the changes in the histogram of type() seen by the jenkin might be simply related to a shift here

**Comment by mariadalfonso** [comment]
please test

**Comment by mariadalfonso** [comment]
+xpog

changes as expected on BeanSpot and Rho

**Comment by perrotta** [comment]
+1
- New variables added to nanoAod output

**Pull request merged**